cineast-core/src/main/java/org/vitrivr/cineast/core/render/lwjgl/util/fsm/abstractworker/StateProviderAnnotationParser.java

```diff
@@ -0,0 +1,178 @@
+package org.vitrivr.cineast.core.render.lwjgl.util.fsm.abstractworker;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import org.vitrivr.cineast.core.render.lwjgl.util.datatype.Variant;
+import org.vitrivr.cineast.core.render.lwjgl.util.fsm.model.Action;
+import org.vitrivr.cineast.core.render.lwjgl.util.fsm.model.State;
+import org.vitrivr.cineast.core.render.lwjgl.util.fsm.model.Transition;
+
+/**
+ * This class is used to parse the annotations of a state provider
+ * @see StateProviderAnnotationParser#runTransitionMethods(Object, State, State, Transition, Variant)
+ */
+public class StateProviderAnnotationParser {
```

Comment:
Very much not a fan of this sort of stuff. Why would this be needed in the first place?

Reply:
While it might go beyond our current needs, the annotation based state machine framework is used by the renderer, cleanly coded, self contained, and well documented.
If we decide that this offers much more flexibility than we want and a simpler solution is built, we can replace it. For now I recommend we merge this as is.